CMake flac build error with BUILD_SHARED_LIBS=ON

```
[71/118] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\flac\CMakeFiles\flacapp.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1421~1.277\bin\Hostx64\x64\link.exe  src\flac\CMakeFiles\flacapp.dir\analyze.c.obj src\flac\CMakeFiles\flacapp.dir\decode.c.obj src\flac\CMakeFiles\flacapp.dir\encode.c.obj src\flac\CMakeFiles\flacapp.dir\foreign_metadata.c.obj src\flac\CMakeFiles\flacapp.dir\main.c.obj src\flac\CMakeFiles\flacapp.dir\local_string_utils.c.obj src\flac\CMakeFiles\flacapp.dir\utils.c.obj src\flac\CMakeFiles\flacapp.dir\vorbiscomment.c.obj  /out:src\flac\flac.exe /implib:src\flac\flacapp.lib /pdb:src\flac\flac.pdb /version:0.0  /machine:x64 /debug /INCREMENTAL  /subsystem:console  src\share\getopt\getopt.lib src\share\replaygain_synthesis\replaygain_synthesis.lib src\share\utf8\utf8.lib src\share\win_utf8_io\win_utf8_io.lib src\share\grabbag\grabbag.lib src\libFLAC\FLAC.lib D:\vcpkg\installed\x64-windows\debug\lib\ogg.lib src\share\win_utf8_io\win_utf8_io.lib src\share\replaygain_analysis\replaygain_analysis.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
FAILED: src/flac/flac.exe 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\flac\CMakeFiles\flacapp.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1421~1.277\bin\Hostx64\x64\link.exe  src\flac\CMakeFiles\flacapp.dir\analyze.c.obj src\flac\CMakeFiles\flacapp.dir\decode.c.obj src\flac\CMakeFiles\flacapp.dir\encode.c.obj src\flac\CMakeFiles\flacapp.dir\foreign_metadata.c.obj src\flac\CMakeFiles\flacapp.dir\main.c.obj src\flac\CMakeFiles\flacapp.dir\local_string_utils.c.obj src\flac\CMakeFiles\flacapp.dir\utils.c.obj src\flac\CMakeFiles\flacapp.dir\vorbiscomment.c.obj  /out:src\flac\flac.exe /implib:src\flac\flacapp.lib /pdb:src\flac\flac.pdb /version:0.0  /machine:x64 /debug /INCREMENTAL  /subsystem:console  src\share\getopt\getopt.lib src\share\replaygain_synthesis\replaygain_synthesis.lib src\share\utf8\utf8.lib src\share\win_utf8_io\win_utf8_io.lib src\share\grabbag\grabbag.lib src\libFLAC\FLAC.lib D:\vcpkg\installed\x64-windows\debug\lib\ogg.lib src\share\win_utf8_io\win_utf8_io.lib src\share\replaygain_analysis\replaygain_analysis.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK Pass 1: command "C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1421~1.277\bin\Hostx64\x64\link.exe src\flac\CMakeFiles\flacapp.dir\analyze.c.obj src\flac\CMakeFiles\flacapp.dir\decode.c.obj src\flac\CMakeFiles\flacapp.dir\encode.c.obj src\flac\CMakeFiles\flacapp.dir\foreign_metadata.c.obj src\flac\CMakeFiles\flacapp.dir\main.c.obj src\flac\CMakeFiles\flacapp.dir\local_string_utils.c.obj src\flac\CMakeFiles\flacapp.dir\utils.c.obj src\flac\CMakeFiles\flacapp.dir\vorbiscomment.c.obj /out:src\flac\flac.exe /implib:src\flac\flacapp.lib /pdb:src\flac\flac.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console src\share\getopt\getopt.lib src\share\replaygain_synthesis\replaygain_synthesis.lib src\share\utf8\utf8.lib src\share\win_utf8_io\win_utf8_io.lib src\share\grabbag\grabbag.lib src\libFLAC\FLAC.lib D:\vcpkg\installed\x64-windows\debug\lib\ogg.lib src\share\win_utf8_io\win_utf8_io.lib src\share\replaygain_analysis\replaygain_analysis.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\flac\CMakeFiles\flacapp.dir/intermediate.manifest src\flac\CMakeFiles\flacapp.dir/manifest.res" failed (exit code 1120) with the following output:
Microsoft (R) Incremental Linker Version 14.21.27702.2
Copyright (C) Microsoft Corporation.  All rights reserved.

main.c.obj : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
vorbiscomment.c.obj : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
grabbag.lib(picture.c.obj) : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
analyze.c.obj : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
decode.c.obj : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
encode.c.obj : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
foreign_metadata.c.obj : error LNK2001: unresolved external symbol flac_internal_fopen_utf8
decode.c.obj : error LNK2019: unresolved external symbol flac_internal_unlink_utf8 referenced in function DecoderSession_destroy
encode.c.obj : error LNK2001: unresolved external symbol flac_internal_unlink_utf8
main.c.obj : error LNK2001: unresolved external symbol flac_internal_unlink_utf8
grabbag.lib(file.c.obj) : error LNK2001: unresolved external symbol flac_internal_unlink_utf8
main.c.obj : error LNK2019: unresolved external symbol flac_internal_rename_utf8 referenced in function encode_file
win_utf8_io.lib(win_utf8_io.c.obj) : error LNK2019: unresolved external symbol flac_internal_set_utf8_filenames referenced in function get_utf8_argv
win_utf8_io.lib(win_utf8_io.c.obj) : error LNK2019: unresolved external symbol flac_internal_get_utf8_filenames referenced in function CreateFile_utf8
grabbag.lib(file.c.obj) : error LNK2019: unresolved external symbol flac_internal_stat64_utf8 referenced in function grabbag__file_copy_metadata
grabbag.lib(replaygain.c.obj) : error LNK2001: unresolved external symbol flac_internal_stat64_utf8
grabbag.lib(file.c.obj) : error LNK2019: unresolved external symbol flac_internal_chmod_utf8 referenced in function grabbag__file_copy_metadata
grabbag.lib(replaygain.c.obj) : error LNK2001: unresolved external symbol flac_internal_chmod_utf8
grabbag.lib(file.c.obj) : error LNK2019: unresolved external symbol flac_internal_utime_utf8 referenced in function grabbag__file_copy_metadata
src\flac\flac.exe : fatal error LNK1120: 8 unresolved externals
```

@krokoziabla

--- a/src/libFLAC/CMakeLists.txt
+++ b/src/libFLAC/CMakeLists.txt
@@ -87,10 +87,12 @@ add_library(FLAC
     $<$<BOOL:${OGG_FOUND}>:ogg_decoder_aspect.c>
     $<$<BOOL:${OGG_FOUND}>:ogg_encoder_aspect.c>
     $<$<BOOL:${OGG_FOUND}>:ogg_helper.c>
-    $<$<BOOL:${OGG_FOUND}>:ogg_mapping.c>
-    $<$<BOOL:${WIN32}>:windows_unicode_filenames.c>)
+    $<$<BOOL:${OGG_FOUND}>:ogg_mapping.c>)
 if(TARGET FLAC-asm)
     target_sources(FLAC PRIVATE $<TARGET_OBJECTS:FLAC-asm>)
+endif()
+if(WIN32)
+    target_sources(FLAC PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/windows_unicode_filenames.c>)
 endif()
 
 target_compile_definitions(FLAC

